### PR TITLE
Convert YouTube urls to be embeddable 

### PR DESCRIPTION
**Pull request by nathanstitt**
uses methods discussed in https://stackoverflow.com/questions/21607808/convert-a-youtube-video-url-to-embed-code/21607897

### CODE REVIEW

File: src/URLs.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
+const HTTP_PREFIX = /^http(s?):*\/\//i
+
+export function sanitizeURL(url: string | null | undefined): string {
+    if (!url) {
+        return 'http://'
+    }
```
Comment (camliu89) [inline src/URLs.ts:6]: I am wondering if this should return an empty string. Would the editor handle well a link like `http://`?

File: src/URLs.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    if (HTTP_PREFIX.test(url)) {
+        return url
+    }
+    return 'http://' + url
+}
+
+const YOUTUBE = /^.*(youtu.be\/|v\/|u\/\w\/|embed\/|watch\?v=|\&v=)(.*)/;
```
Comment (camliu89) [inline src/URLs.ts:13]: nice regex. :)

File: src/URLs.ts (hunk 1 above)
Comment (nathanstitt) [reply src/URLs.ts:6]: yes good question, I'm not sure at all why it does that.  I'll just return whatever it was there

**Review by camliu89 — APPROVED** [review]
looks good.

**Pull request merged**